support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -50,26 +57,34 @@ class RedemptionController(
   val js = "subscriptionsRedemptionPage.js"
   val css = "digitalSubscriptionCheckoutPage.css" //TODO: Don't need this?
 
-  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = googleAuthAction.async {
+  val getCorporateCustomer = new GetCorporateCustomer(dynamoLookup)
+  val testUserFromRequest = new TestUserFromRequest(identityService, testUsers)
+
+  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = (googleAuthAction andThen maybeAuthenticatedAction()).async {
     implicit request =>
-      getCorporateCustomer(redemptionCode).fold(Some.apply, _ => None).map(
-        maybeError =>
-          Ok(subscriptionRedemptionForm(
-            title = title,
-            mainElement = id,
-            js = js,
-            css = css,
-            fontLoaderBundle = fontLoaderBundle,
-            csrf = None,
-            uatMode = false,
-            stage = "checkout",
-            redemptionCode = redemptionCode,
-            maybeRedemptionError = maybeError,
-            user = None,
-            submitted = false
-          )).withHeaders(CacheControl.noCache)
-      )
-    }
+      for {
+        whyIsTestUser <- testUserFromRequest.getTestUserStatus(request)
```

Comment:
the comment has gone now @rupertbates but we could call this `testUserStatus` perhaps.  either/or we could consider a named ADT rather than leaning on Option in this unexpected way.